Cleaned the yum cache after the yum update for ServerJRE8

aureliogrb opened the pull request:
Also removed unneeded clean up from the builder image and changed from yum/microdnf clean all to rm -rf /var/cache/yum 

Signed-off-by: Aurelio Garcia-Ribeyro <aurelio.garciaribeyro@oracle.com>

Djelibeybi commented:
On Oracle Linux 8, you need to run `rm -rf /var/cache/dnf` instead.

Djelibeybi commented on OracleJava/11/Dockerfile.8-slim:
```diff
@@ -73,7 +72,7 @@ RUN set -eux; \
 		freetype fontconfig \
 	; \
-	microdnf clean all; \
+	rm -rf /var/cache/yum; \
```
This should be `/var/cache/dnf`

aureliogrb replied:
Fixed.  Thanks

aureliogrb replied:
I will use OL8 for JDK 17+ but will keep the current images on OL8 Slim to avoid surprising existing users.

Djelibeybi commented on OracleJava/8/Dockerfile.8-slim:
```diff
@@ -70,6 +69,7 @@ COPY --from=builder $JAVA_HOME $JAVA_HOME
 
 ##
 RUN	microdnf -y update; \
+	rm -rf /var/cache/yum; \
```
This should be `/var/cache/dnf`

aureliogrb replied:
Fixed.  Thanks

Djelibeybi reviewed (dismissed):
You should switch to using the `oraclelinux:8` image and change from `microdnf` to `dnf` as well. We only recommend using the `8-slim` image with microservices that are statically compiled and don't require any additional packages.

Djelibeybi reviewed (changes requested):
You still need to change `FROM oraclelinux:8-slim` to `FROM oraclelinux:8` in all the OL8-based images.

aureliogrb commented:
I will change to oraclelinux:8 for JDK 17 and later but don't think we should do so for the existing images that are already being offered in 8-slim.

Djelibeybi commented:
You can change the existing images to `FROM ghcr.io/oracle/oraclelinux8-compat:8-slim` so that they get the compatibility base image. Essentially, this is `oraclelinux:8` with both `dnf` and `microdnf` installed so any downstream CI/CD that relies on `microdnf` will continue to work.

aureliogrb commented:
Updated the oraclelinux:8-slim references to ghcr.io/oracle/oraclelinux8-compat:8-slim.  Since that image has both dnf and microdnf changed scripts to use dnf.  Also removed some unnecessary echos at the end of the build file and unneded xshare dump for versions > 11.
